src/ferny/ssh_errors.py

```diff
@@ -41,6 +41,10 @@ def __init__(self, match: Match, stderr: str) -> None:
         self.message = match.group(0)
 
 
+class SshInvalidHostnameError(SshError):
+    PATTERN = re.compile(r'^hostname contains invalid characters', re.I)
```

Comment:
I deliberately did *not* match a `$` here -- future versions may well extend the error message to show the invalid character or some additional help. This is the most robust compromise IMHO.

Reply:
This needs a testcase added in the `STDERR_MESSAGES` section of `test_regexp.py`

Reply:
Done.